src/hydra/dataProvider.ts

```diff
@@ -222,44 +248,49 @@ function dataProvider(
     );
   }
 
-  /** @type {Api} */
-  let apiSchema;
+  let apiSchema: Api & { resources: Resource[] };
 
   // store mercure subscriptions
-  const subscriptions = {};
+  const subscriptions: { [key: string]: MercureSubscription } = {};
 
-  /**
-   * @param {Resource} resource
-   * @param {Object} data
-   *
-   * @returns {Promise<Object>}
-   */
-  const convertReactAdminDataToHydraData = (resource, data = {}) => {
-    const fieldData: any[] = [];
-    resource.fields.forEach(({ name, reference, normalizeData }) => {
-      if (!(name in data)) {
-        return;
-      }
+  const convertReactAdminDataToHydraData = (
+    resource: Resource,
+    data: { [key: string]: unknown } = {},
+  ) => {
+    const fieldData: { [key: string]: unknown } = {};
+    if (resource.fields) {
+      (
+        resource.fields as (Field & {
+          normalizeData: (data: unknown) => unknown;
+        })[]
+      ).forEach(({ name, reference, normalizeData }) => {
+        if (!(name in data)) {
+          return;
+        }
 
-      if (reference && data[name] === '') {
-        data[name] = null;
-        return;
-      }
+        if (reference && data[name] === '') {
+          data[name] = null;
+          return;
+        }
 
-      if (undefined === normalizeData) {
-        return;
-      }
+        if (undefined === normalizeData) {
+          return;
+        }
 
-      fieldData[name] = normalizeData(data[name]);
-    });
+        fieldData[name] = normalizeData(data[name]);
+      });
+    }
 
-    const fieldDataKeys: any = Object.keys(fieldData);
+    const fieldDataKeys = Object.keys(fieldData);
     const fieldDataValues = Object.values(fieldData);
 
     return Promise.all(fieldDataValues).then((fieldData) => {
-      const object = {};
+      const object: { [key: string]: unknown } = {};
```

Comment:
```suggestion
      const object: Record<string: unknown> = {};
```
Apply to everywhere you have `{ [key: string]: something }` (`Record<string, something>`)